### PR TITLE
Sharing: add QR Code to original sharing popup

### PR DESCRIPTION
**Pull request by breville**
This adds a QR Code to the original sharing popup that shows at the end of many levels.  It contains the URL that the user might choose to share.

It means the "Send to phone" button is now shown for all users, not just US users, but only US users will see the phone number fields, while all users will see the QR Code.

I took the time to adjust the layout of the popup a bit, and it's now only 6 pixels taller than it was before, when the phone number is showing, as seen in the first two screenshots.

### before - showing

Screenshot 2020-03-06 16 53 41

### after - showing

Screenshot 2020-03-06 14 30 23

### after - showing for non-US user

Screenshot 2020-03-06 16 31 25

### after - hidden

Screenshot 2020-03-06 14 41 43

### after - no image - showing

Screenshot 2020-03-06 14 40 00

### after - no image - hidden

Screenshot 2020-03-06 14 39 44

### CODE REVIEW

File: apps/src/templates/sharing.html.ejs [hunk 1]
```diff
@@ -55,22 +55,35 @@
...
 <% } %>
 
 </div>
+
+<% /* Always show the "Send to phone" button, and when it's clicked to toggle
+    * the section below into view, the QR Code, but only show the phone number
+    * in that section below when options.sendToPhone is true, which is done
+    * for US users. */ %>
+<div id="send-to-phone" style="display: none">
+  <% if (options.sendToPhone) { %>
```
Comment (islemaster) [inline apps/src/templates/sharing.html.ejs]: I'd be nice to rename or replace this option end-to-end to something more specific now that there are two "send to phone" features; maybe `smsToPhone` or even just `inUS`

Comment (breville) [reply apps/src/templates/sharing.html.ejs]: You're right.  I got nervous about changing the plumbing on such old code, but I think I've got it now...

**Review by maddiedierker — APPROVED** [review]

File: apps/src/templates/sharing.html.ejs [hunk 2]
```diff
@@ -55,22 +55,35 @@
...
+  <% if (options.isUS) { %>
+    <div id="send-to-phone-left">
+      <label for="phone">Enter a US phone number:</label>
+      <input type="tel" id="phone" name="phone" />
+      <button id="phone-submit" onClick="return false;">Send</button>
+      <div id="phone-charges">A text message will be sent via <a href="http://twilio.com" target="_blank">Twilio</a>. Charges may apply to the recipient.</div>
+    </div>
+  <% } %>
+  <div id="send-to-phone-right">
+    <label>Scan this code with your phone camera:</label>
```
Comment (Hamms) [inline apps/src/templates/sharing.html.ejs:86]: can we translate this?

Comment (breville) [reply apps/src/templates/sharing.html.ejs:86]: Oops yes.  Good catch.  Updated in https://github.com/code-dot-org/code-dot-org/pull/33902.

Comment (Hamms) [reply apps/src/templates/sharing.html.ejs:86]: thanks!

**Pull request merged**